### PR TITLE
refactor: address comments from #3415: Prepare for changes in ZIP-244

### PR DESCRIPTION
**Pull request by conradoplg**
## Motivation

https://github.com/ZcashFoundation/zebra/pull/3415/ was accidentally merged without addressing review comments. This PR addresses them.

### Reviewer Checklist

  - [x] Code implements Specs and Designs
  - [x] Tests for Expected Behaviour
  - [ ] Tests for Errors

### CODE REVIEW

File: zebra-consensus/src/transaction.rs [hunk 1]
```diff
@@ -457,31 +457,31 @@ where
...
-                }
-                transparent::Input::Coinbase { .. } => continue,
+                        unreachable!("AwaitUtxo always responds with Utxo")
+                    }
+                };
+                tracing::trace!(?utxo, "got UTXO");
+                spent_outputs.push(utxo.output.clone());
+                spent_utxos.insert(*outpoint, utxo);
+            } else {
+                continue;
```
Comment (jvff) [inline zebra-consensus/src/transaction.rs:479]: Optional nit: this feels redundant to me :thinking: 

```suggestion
```

File: zebra-consensus/src/transaction.rs [hunk 2]
```diff
@@ -457,31 +457,31 @@ where
...
-                    tracing::trace!("awaiting outpoint lookup");
-                    let utxo = if let Some(output) = known_utxos.get(outpoint) {
-                        tracing::trace!("UXTO in known_utxos, discarding query");
-                        output.utxo.clone()
+            if let transparent::Input::PrevOut {
+                outpoint,
+                unlock_script: _,
+                sequence: _,
+            } = input
+            {
```
Comment (jvff) [inline zebra-consensus/src/transaction.rs]: Optional: removing the two unused fields makes this a bit more concise, but it's subjective whether it's better or not :sweat_smile: 

```suggestion
            if let transparent::Input::PrevOut { outpoint, .. } = input {
```

**Review by jvff — COMMENTED** [review]
Consider this approved. I'm not marking it as approved in case it gets merged again :sweat_smile: 

I left two optional suggestions, but feel free to merge without them :)

File: zebra-consensus/src/transaction.rs (hunk 2 above)
Comment (teor2345) [reply zebra-consensus/src/transaction.rs]: I think shorter is more readable, but I don't mind either way.

File: zebra-consensus/src/transaction.rs (hunk 1 above)
Comment (dconnolly) [reply zebra-consensus/src/transaction.rs:479]: I kinda appreciate the explicit continue in the loop 😅

**Review by dconnolly — APPROVED** [review]
LGTM!

Let's see if Mergify will merge with at least one unresolved conversation 👀

**Pull request merged**